pkg/storage/storage.go

```diff
@@ -241,129 +230,129 @@ func filterTuples(tuples []*openfgapb.TupleKey, targetObject, targetRelation str
+func (c *combinedTupleReader) ReadUsersetTuples(
 	ctx context.Context,
 	store string,
 	filter ReadUsersetTuplesFilter,
 ) (TupleIterator, error) {
 
-	var iter1 TupleIterator = &emptyTupleIterator{}
+	var usersetTuples []*openfgapb.Tuple
 
-	var filteredTuples []*openfgapb.Tuple
-
-	tuples, ok := ContextualTuplesFromContext(ctx)
-	if ok {
-		for _, t := range filterTuples(tuples, filter.Object, filter.Relation) {
-			if tuple.GetUserTypeFromUser(t.GetKey().GetUser()) == tuple.UserSet {
-				filteredTuples = append(filteredTuples, t)
-			}
+	for _, t := range filterTuples(c.contextualTuples, filter.Object, filter.Relation) {
+		if tuple.GetUserTypeFromUser(t.GetKey().GetUser()) == tuple.UserSet {
+			usersetTuples = append(usersetTuples, t)
 		}
-
-		iter1 = NewStaticTupleIterator(filteredTuples)
 	}
 
-	iter2, err := b.wrapped.ReadUsersetTuples(ctx, store, filter)
+	iter1 := NewStaticTupleIterator(usersetTuples)
+
+	iter2, err := c.wrapped.ReadUsersetTuples(ctx, store, filter)
 	if err != nil {
 		return nil, err
 	}
 
 	return NewCombinedIterator(iter1, iter2), nil
 }
 
-func (b *ctxRelationshipTupleReader) ReadStartingWithUser(
+func (c *combinedTupleReader) ReadStartingWithUser(
 	ctx context.Context,
 	store string,
 	filter ReadStartingWithUserFilter,
 ) (TupleIterator, error) {
 
-	var iter1 TupleIterator = &emptyTupleIterator{}
-
-	tuples, ok := ContextualTuplesFromContext(ctx)
-	if ok {
+	var filteredTuples []*openfgapb.Tuple
+	for _, t := range c.contextualTuples {
+		if tuple.GetType(t.GetObject()) != filter.ObjectType {
+			continue
+		}
 
-		var filteredTuples []*openfgapb.Tuple
-		for _, t := range tuples {
-			if tuple.GetType(t.GetObject()) != filter.ObjectType {
-				continue
-			}
+		if t.GetRelation() != filter.Relation {
+			continue
+		}
 
-			if t.GetRelation() != filter.Relation {
-				continue
+		for _, u := range filter.UserFilter {
+			targetUser := u.GetObject()
+			if u.GetRelation() != "" {
+				targetUser = tuple.ToObjectRelationString(targetUser, u.GetRelation())
 			}
 
-			for _, u := range filter.UserFilter {
-				targetUser := u.GetObject()
-				if u.GetRelation() != "" {
-					targetUser = tuple.ToObjectRelationString(targetUser, u.GetRelation())
-				}
-
-				if t.GetUser() == targetUser {
-					filteredTuples = append(filteredTuples, &openfgapb.Tuple{
-						Key: t,
-					})
-				}
+			if t.GetUser() == targetUser {
+				filteredTuples = append(filteredTuples, &openfgapb.Tuple{
+					Key: t,
+				})
 			}
 		}
-
-		iter1 = NewStaticTupleIterator(filteredTuples)
 	}
 
-	iter2, err := b.wrapped.ReadStartingWithUser(ctx, store, filter)
+	iter1 := NewStaticTupleIterator(filteredTuples)
+
+	iter2, err := c.wrapped.ReadStartingWithUser(ctx, store, filter)
 	if err != nil {
 		return nil, err
 	}
 
 	return NewCombinedIterator(iter1, iter2), nil
 }
+
+func (c *combinedTupleReader) ListObjectsByType(ctx context.Context, store string, objectType string) (ObjectIterator, error) {
```

Comment:
mind adding description for exportable function (i.e., to satisfy godoc)?

Reply:
The documentation for all these methods is in the interface declaration: https://github.com/openfga/openfga/blob/c4a8d8f5ad36c02d13c0470cc68e3b5c204bac8d/pkg/storage/storage.go#L103